### PR TITLE
Get all mvp data

### PR DESCRIPTION
**Pull request by dynamictulip**
First changes to "get all the data" - this covers adding new EF models for the GIAS tables and using them to populate the new Academy type.

[User Story 146083](https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_workitems/edit/146083): Build: Get all the data we need from Academies DB for MVP

## Changes

**Included**

- Added new data types for UI
- Added EF models for GIAS tables
- Extracted "helper" methods into extension classes and rebranded `TrustHelper` as `TrustFactory`
- Change file structure in `DfE.FindInformationAcademiesTrusts.Data.AcademiesDb`
- Create Academy type in `TrustProvider` from `Establishment`s and `GroupLink`s associated with a `Group`

**Not included**

- Tests for new `AgeRange` record -> waiting on pagination work to be merged into main as this provides a unit test project for the `DfE.FindInformationAcademiesTrusts.Data` project

## Checklist

- [n/a] Deploy this branch to the test environment for manual testing once comments have been resolved and checks have passed 
- [x] Attach this pull request to the appropriate user story in Azure DevOps
- [n/a] Update the ADR decision log if needed

### CODE REVIEW

File: DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Factories/AcademyFactory.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    Academy CreateAcademyFrom(GroupLink gl, Establishment establishment);
+}
+
+public class AcademyFactory : IAcademyFactory
+{
+    public Academy CreateAcademyFrom(GroupLink gl, Establishment establishment)
+    {
+        return new Academy(
+            establishment.Urn,
+            DateTime.ParseExact(gl.JoinedDate!, "dd/MM/yyyy", CultureInfo.InvariantCulture),
```
Comment (lookupdaily) [inline DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Factories/AcademyFactory.cs:18]: can you use your new date extension here?

**Review by lookupdaily — APPROVED** [review]
Looking very shiny 🥇 

Love the extension methods! 

Could potentially squash that last commit into your 'rename to factories' one?

File: DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Factories/AcademyFactory.cs (hunk 1 above)
Comment (dynamictulip) [reply DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Factories/AcademyFactory.cs:18]: This one is a non-nullable date so doesn't need the extension method :-)

**Comment by dynamictulip** [comment]
> Looking very shiny 🥇
> 
> Love the extension methods!
> 
> Could potentially squash that last commit into your 'rename to factories' one?

Good idea!

**Pull request merged**